test/models/drive_test.dart

```diff
@@ -0,0 +1,489 @@
+        ],
+      );
+      expect(drive.pendingRides, [ride1, ride2]);
+    });
+  });
+  group('Drive.userHasDriveAtTimeRange', () {
+    test('has drive in time range', () async {
+      when.call(driveProcessor.processUrl(any)).thenReturn(jsonEncode([
+            DriveFactory()
+                .generateFake(
+                  driverId: 2,
+                  startTime: DateTime.now().add(const Duration(hours: 2)),
+                  endTime: DateTime.now().add(const Duration(hours: 4)),
+                )
+                .toJsonForApi()
+          ]));
+      expect(
+          await Drive.userHasDriveAtTimeRange(
+              DateTimeRange(start: DateTime.now(), end: DateTime.now().add(const Duration(hours: 10))), 2),
+          true);
+    });
+    test('has no drive at time range', () async {
+      when.call(driveProcessor.processUrl(any)).thenReturn(jsonEncode([
+            DriveFactory()
+                .generateFake(
+                  driverId: 2,
+                  startTime: DateTime.now().add(const Duration(hours: 8)),
+                  endTime: DateTime.now().add(const Duration(hours: 10)),
+                )
+                .toJsonForApi(),
+            DriveFactory()
+                .generateFake(
+                  driverId: 2,
+                  startTime: DateTime.now().add(const Duration(hours: 14)),
+                  endTime: DateTime.now().add(const Duration(hours: 16)),
+                )
+                .toJsonForApi(),
+          ]));
+      expect(
+          await Drive.userHasDriveAtTimeRange(
+              DateTimeRange(
+                start: DateTime.now().add(const Duration(hours: 11)),
+                end: DateTime.now().add(const Duration(hours: 13)),
+              ),
+              2),
+          false);
+    });
+    test('drive overlaps with time range', () async {
+      when.call(driveProcessor.processUrl(any)).thenReturn(jsonEncode([
+            DriveFactory()
+                .generateFake(
+                  driverId: 2,
+                  startTime: DateTime.now().add(const Duration(hours: 7)),
+                  endTime: DateTime.now().add(const Duration(hours: 10)),
+                )
+                .toJsonForApi(),
+          ]));
+      expect(
+          await Drive.userHasDriveAtTimeRange(
+              DateTimeRange(
+                start: DateTime.now().add(const Duration(hours: 6)),
+                end: DateTime.now().add(const Duration(hours: 8)),
+              ),
+              2),
+          true);
+    });
+  });
+  group('Drive.getMaxUsedSeats', () {
+    test('can handle null', () async {
+      final Drive drive = DriveFactory().generateFake(createDependencies: false);
+      expect(drive.getMaxUsedSeats(), null);
+    });
+    test('no seats taken', () async {
+      final Drive drive = DriveFactory().generateFake(seats: 3, rides: <Ride>[
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
```

Comment:
Could you please continue using `DateTime.now()` here and everywhere else? I think this is really illegible

Reply:
Think you forgot that one because GitHub kind of hides it ;)